Adding automation new issues in project board

mijovic opened the pull request.

mijovic committed:
Adding automation for new, assigned and stale issues in project board

axic commented on .github/workflows/assigned_issues.yml:
```diff
@@ -0,0 +1,15 @@
+name: Move new issues into "New issues" in "Solidity" project board
```
This seems to be for assignment?

mijovic replied:
Copy paste error... Fixing now

mijovic commented on .github/workflows/stale_issues.yml:
```diff
@@ -0,0 +1,17 @@
+Name: Comment and mark stale issues
+
+on:
+  schedule:
+  - cron: "0 0 * * *"
+
+jobs:
+  stale:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/stale@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        stale-issue-message: 'Issue has been inactive for too long. Consider closing or moving to icebox. This message is automated.'
+        stale-issue-label: 'no-issue-activity'
```
Needs to exclude issues that are already in Icebox

mijovic commented on .github/workflows/assigned_issues.yml:
```diff
@@ -0,0 +1,15 @@
+        with:
+          project: Solidity
+          column: In progress
+          repo-token: ${{ secrets.ASSIGNED_ISSUES_TOKEN }}
```
We can use `GITHUB_TOKEN` as well

mijovic commented:
It is updated to work with new issues only

chriseth commented:
Will this "just work" once it is merged?

mijovic commented:
> Will this "just work" once it is merged?

It should "just work". But let me check once more on my repo so I can be 100% sure.

mijovic commented:
Changed to different github action, as the one I used for first needed more permissions than `GITHUB_TOKEN` is giving.
You can find here more about `GITHUB_TOKEN` https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token

Updated PR and it will sort issues to `Solidity` project automatically once merged (and default column is `New issues`).

chriseth commented:
Looking at skeet70/default-project-board-action I find it weird that it has to perform additional requests to retrieve the issue and project id. I'm wondering if we were better of just writing those scripts ourselves...

mijovic commented:
> Looking at skeet70/default-project-board-action I find it weird that it has to perform additional requests to retrieve the issue and project id. I'm wondering if we were better of just writing those scripts ourselves...

I think yes, as it is really simple thing. That will allow us to do more custom things later.
But than, where it should be done. I think solidity repo is not the place where we want this code :)

chriseth commented:
I'm mainly worried that we are exposing the github token to all the npm dependencies this pulls in.

leonardoalt commented:
Was there a decision whether to use this or write our own?

chriseth commented:
I would feel more comfortable with writing our own - I don't think it is more complicated than the scripts we already have, and it can be tweaked more easily.

mijovic commented:
> I would feel more comfortable with writing our own - I don't think it is more complicated than the scripts we already have, and it can be tweaked more easily.

Let's close this one than and start writing our own.
